In [1]:
import pandas as pd

In [2]:
from pandas import Series, DataFrame

In [3]:
import matplotlib.pyplot as plt

In [4]:
import seaborn as sns

In [5]:
from sklearn import datasets

In [6]:
from sklearn import metrics

In [7]:
from sklearn.naive_bayes import GaussianNB

In [8]:
from sklearn.cross_validation import train_test_split

# Supervised Learning - Naive Bayes Classifiers

This project will be a practice of using the very practical Naive Bayes algorithm for performing Multi-class Classification. Serving as the independent feature model, the Naive Bayes probability model will be used along with a decision rule for deciding which class of flower is most probable. The Maximum A Posteriori (MAP) estimation will be used for picking the most probable hypothesis/flower class.

------

Below, the dataset will be loaded as iris = datasets.load_iris(). 

In [9]:
iris = datasets.load_iris()

Next, the features (X) will be grabbed.

In [10]:
X = iris.data

Next, the target (Y) will be grabbed.

In [11]:
Y = iris.target

Below, a model will be made by calling GaussianNB() with no additional arguments.

In [12]:
model = GaussianNB()

Next, the model will be separated into training and testing sets by using train_test_split from the cross validation library in scikit-learn. The data will be split into training and testing for both the features and the target. 

In [13]:
XTrain, XTest, YTrain, YTest = train_test_split(X, Y)

The model will be fit below using the training datasets. To do that, the training data (both the features and the classes) will be passed through model.fit().

In [14]:
model.fit(XTrain, YTrain)

GaussianNB()

Next, the outcomes from the testing X features will be compared to the Y test classes. The model.predict() method is what will be used to predict the outcomes from the testing set. An object will be created named  “predicted” that will be set equal to model.predict() with the “XTest” data passed through it.

In [15]:
predicted = model.predict(XTest)

YTest has the data with the actual expected outcomes. Below, an object named “expected” will be created and set to that data.

In [16]:
expected = YTest

The print function (with expected versus predicted passed) will be used in order to see the metrics for performance.

In [17]:
print (metrics.accuracy_score(expected, predicted))

0.921052631579


It seems that 92% is not actually a bad accuracy, considering the naive assumption that all of those features were independent of each other. This result can change though, based on how the randomness of the train_test_split occurred. 